src/ini.pest

```diff
@@ -1,13 +1,10 @@
-WHITESPACE = _{ " " }
-char = { ASCII_ALPHANUMERIC | "." | "_" | "/" }
-name = @{ char+ }
-value = @{ char* }
+ini = { SOI ~ (section | pair | comment)* ~ EOI }
 
-section = { "[" ~ name ~ "]" }
-property = { name ~ "=" ~ value }
+section = { "[" ~ key ~ "]" }
+pair = { key ~ "=" ~ value }
+comment = { (";" | "#" ~ value }
 
-file = {
-	SOI ~
-	((section | property)? ~ NEWLINE)* ~
-	EOI
-}
+key = @{ (!("]" | "=" | WHITESPACE) ~ ANY)+ }
```

Comment:
There's no reason that the key shouldn't be able to have a `]` in both the section name as well as the pair key.

Reply:
You're right - this currently does not work with this grammar without lookahead. I have to think how to implement it without messing the grammar to much up.